n2kMapper.js

```diff
@@ -65,45 +68,49 @@ function getValue (n2k, theMapping, state) {
 
 var toValuesArray = function (theMappings, n2k, state) {
   if (n2k.fields && typeof theMappings !== 'undefined') {
-    return theMappings
-      .filter(function (theMapping) {
-        try {
-          return (
-            typeof theMapping.filter === 'undefined' ||
-            theMapping.filter(n2k, state)
-          )
-        } catch (ex) {
-          process.stderr.write(ex + ' ' + n2k)
-          return false
-        }
-      })
-      .map(function (theMapping) {
-        try {
-          var path =
-            typeof theMapping.node === 'function'
-              ? theMapping.node(n2k, state)
-              : theMapping.node
-          var value =
-            typeof theMapping.source === 'function'
-              ? theMapping.source(n2k, state)
-              : getValue(n2k, theMapping, state)
-          var allowNull =
-            typeof theMapping.allowNull !== 'undefined' &&
+    if (typeof theMappings === 'function') {
+      return theMappings(n2k, state)
+    } else {
+      return theMappings
+        .filter(function (theMapping) {
+          try {
+            return (
+              typeof theMapping.filter === 'undefined' ||
+              theMapping.filter(n2k, state)
+            )
+          } catch (ex) {
+            process.stderr.write(ex + ' ' + n2k)
+            return false
+          }
+        })
+        .map(function (theMapping) {
```

Comment:
Now the code assumes that there is only one mapping if the mapping is a function.

If the test for function was down here you could mix traditional mappings and functions, theMappings would always be an array and the code would have branching in just one spot.